### PR TITLE
296 implement 5X protocol for message transfer

### PR DESCRIPTION
**Pull request by LucasTesnier**
Update notion to include the 5X protocol part.

Add inside the Receiver class the holding of protocol 50.

Create two utility function inside Communicator classes -> CreateAProtocol50 (Pseudo + MessageContent), ReceiveAProtocol50 : {pseudo + messageContent}

Add unit testing for protocol 50 and utility function.

Prepare the code for Pseudo and Password size refacto.

### CODE REVIEW

**Review by 31Nathan — DISMISSED** [review]
LGTM

File: libs/Communicator/Communicator.hpp [hunk 1]
```diff
@@ -120,6 +120,17 @@ namespace communicator_lib
         /// @throw an error when no server can be found (Not in a client communicator), throw a NetworkError
         unsigned short getServerEndpointId(void);
 
+        /// @brief Utilitary function used for send a message with a protocol 50.
```
Comment (osvegn) [inline libs/Communicator/Communicator.hpp:123]: used to send

File: libs/Communicator/Communicator.hpp [hunk 2]
```diff
@@ -120,6 +120,17 @@ namespace communicator_lib
...
+        /// @param pseudo The pseudonyme of the author
+        /// @param messageContent Content of the message
+        /// @param destination of the message
+        void utilitarySendChatMessage(std::string pseudo, std::string messageContent, std::vector<unsigned short> destination);
+
+        /// @brief Utilitary function used to extract a message received with a protocol 50
```
Comment (osvegn) [inline libs/Communicator/Communicator.hpp:129]: received by, I think

**Review by osvegn — CHANGES_REQUESTED** [review]
Minor changes to do

File: libs/Communicator/Communicator.hpp (hunk 2 above)
Comment (LucasTesnier) [reply libs/Communicator/Communicator.hpp]: lgtm

**Review by osvegn — APPROVED** [review]
Look good

**Review by 31Nathan — APPROVED** [review]

**Pull request merged**